### PR TITLE
optimize dynamic gas price calculation

### PR DESCRIPTION
**Pull request by LeoGuo621**
1. Periodically calculate dynamic GP according to gpo.UpdatePeriod
2. Add default configuration for it
3. Only sequencer calculate dynamic GP

### CODE REVIEW

File: cmd/rpcdaemon/commands/eth_api.go [hunk 1]
```diff
@@ -384,7 +385,9 @@ func NewEthAPI(base *BaseAPI, db kv.RoDB, eth rpchelper.ApiBackend, txPool txpoo
 	}
 
 	// For X Layer
-	apii.runL2GasPricerForXLayer()
+	if sequencer.IsSequencer() {
+		apii.runL2GasPricerForXLayer()
+	}
```
Comment (sieniven) [inline cmd/rpcdaemon/commands/eth_api.go:390]: Do we not want to run the X Layer gas pricer on RPC node?

File: eth/gasprice/gaspricecfg/config_xlayer.go [hunk 2]
```diff
@@ -34,7 +34,17 @@ type XLayerConfig struct {
 
 var (
 	DefaultXLayerConfig = XLayerConfig{
-		Type: DefaultType,
+		Type:                            DefaultType,
```
Comment (sieniven) [inline eth/gasprice/gaspricecfg/config_xlayer.go:37]: Fmt

File: cmd/rpcdaemon/commands/eth_api.go (hunk 1 above)
Comment (LeoGuo621) [reply cmd/rpcdaemon/commands/eth_api.go:390]: Actually, eth_gasPrice requests for the RPC nodes are all redirected to the Sequencer node (via zkevm.l2-sequencer-rpc-url). So I think RPC node should not run XLayer gas pricer.

Comment (sieniven) [reply cmd/rpcdaemon/commands/eth_api.go:390]: Sounds good. Can we add a quick comment here for better readability please?

Comment (LeoGuo621) [reply cmd/rpcdaemon/commands/eth_api.go:390]: Sure

**Review by sieniven — APPROVED** [review]

**Pull request merged**